python/pyarrow/tests/test_compute.py

```diff
@@ -536,6 +537,24 @@ def test_slice_compatibility():
                                                start, stop, step) == result
 
 
+def test_binary_slice_compatibility():
+    arr = pa.array((el.encode('ascii')
+                   for el in ["", "a", "ab", "abc", "abcd", "abcde"]))
```

Comment:
Can write this in more idiomatic way. Also, it's nicer with some non-ASCII data:
```suggestion
    arr = pa.array([b"", b"a", b"a\xff", b"a\xffc", b"a\xffcd", b"a\xffcde"])
```